ArcGISRuntimeSDKQt_CppSamples/EditData/EditWithBranchVersioning/EditWithBranchVersioning.qml

```diff
@@ -0,0 +1,348 @@
+// [WriteFile Name=EditWithBranchVersioning, Category=EditData]
+// [Legal]
+// Copyright 2020 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.6
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+import QtGraphicalEffects 1.0
+import Esri.Samples 1.0
+import Esri.ArcGISRuntime.Toolkit.Dialogs 100.10
+import Esri.ArcGISRuntime.Toolkit.Controls 100.10
+
+Item {
+
+    readonly property var featAttributes: ["Affected", "Destroyed", "Inaccessible", "Minor", "Major"]
+    readonly property var versionAccessModel: ["Public", "Protected", "Private"]
+
+    // add a mapView component
+    MapView {
+        id: view
+        anchors.fill: parent
+
+        Callout {
+            id: callout
+            borderWidth: 1
+            borderColor: "lightgrey"
+            calloutData: model.mapView.calloutData
+            leaderPosition: leaderPositionEnum.Automatic
+            onAccessoryButtonClicked: {
+                for (let i=0; i < featAttributes.length; i++) {
+                    if (model.currentTypeDamage === featAttributes[i]) {
+                        typeDmgCombo.currentIndex = i;
+                        break;
+                    }
+                }
+                updateWindow.visible = true;
+            }
+        }
+
+        Button {
+            id: createVersionBtn
+            text: qsTr("Create Version")
+            anchors {
+                left: parent.left
+                top: parent.top
+                margins: 3
+            }
+            enabled: !busyIndicator.visible
+
+            onClicked: {
+                if (text === qsTr("Create Version")) {
+                    createVersionWindow.visible = true;
+                    callout.dismiss();
+                    updateWindow.visible = false;
+                } else if (text === qsTr("Switch to default version")) {
+                    text = qsTr("Switch to created version")
```

Comment:
This logic for changing button text looks like it could be more formalized using `State` blocks: https://doc.qt.io/qt-5/qml-qtquick-state.html

I'd have a think and see what else (beyond just button text) that culd be grouped together as property changes under one state heading.

Reply:
@anmacdonald 
I really do like this suggestion but after a little testing it would require quite a bit of reworking on the logic of when this buttons text changes and when it's disabled etc.. I will definitely keep this in mind going forward but would prefer to leave this as is as it's not the primary purpose of the sample.